src/command/Menus.js

```diff
@@ -1057,37 +1224,70 @@ define(function (require, exports, module) {
             escapedId = StringUtils.jQueryIdEscape(this.id),
             $menuAnchor = $("#" + escapedId),
             $menuWindow = $("#" + escapedId + " > ul"),
-            posTop  = mouseOrLocation.pageY,
-            posLeft = mouseOrLocation.pageX;
+            posTop,
+            posLeft;
 
         // only show context menu if it has menu items
         if ($menuWindow.children().length <= 0) {
             return;
         }
 
-        this.trigger("beforeContextMenuOpen");
-
-        // close all other dropdowns
-        closeAll();
 
         // adjust positioning so menu is not clipped off bottom or right
-        var elementRect = {
+        if (this.parentMenuItem) { // If context menu is a submenu
+
+            this.trigger("beforeSubMenuOpen");
```

Comment:
It will be better to trigger this event before executing line 1231. So wecan add submenu using this event.